styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -279,6 +297,18 @@ public WorkflowResource(Storage storage, WorkflowValidator workflowValidator,
     return Response.forPayload(data);
   }
 
+  /**
+   * Calculate an offset suitable for getting the {@code limit} latest instances of a workflow.
+   */
+  private String calculateTailOffset(int limit, Schedule schedule) {
+    // TODO: Optimize away for loop using Schedule.WellKnown
+    Instant offset = Instant.now();
```

Comment:
Is `now()` a good start? We could use `nextNaturalTrigger` in `WorkflowState` to be more accurate.